### DISCUSSION TITLE
ScriptReference/Playables.PlayableDirector.Pause

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#581] [Notes]
# ScriptReference/Playables.PlayableDirector.Pause

https://docs.unity3d.com/ScriptReference/Playables.PlayableDirector.Pause.html

**Comment by radiatoryang** [comment]
Note that PlayableDirector.Pause() pauses execution only in a weird abstract code sense, and may feel broken when you try to use it

To actually pause a Timeline, like you probably expect, make sure to use this official hack from see https://forum.unity.com/threads/pausing-timeline-playback.473652/

```// actually pause the Timeline
director.playableGraph.GetRootPlayable(0).SetSpeed(0);
director.Pause();

// actually resume the Timeline
director.time = director.time; // refreshes an internal timer thing for certain track types, like Audio
director.playableGraph.GetRootPlayable(0).SetSpeed(1);
director.Resume();
